src/crystal/system/file.cr

```diff
@@ -6,6 +6,10 @@ module Crystal::System::File
   # calls.
   private def self.open_flag(mode : ::File::Mode)
     flags = 0
+    if mode.read? && mode.write?
+      flags |= LibC::O_RDWR
+      mode ^= ::File::Mode::Read | ::File::Mode::Write
+    end
```

Comment:
I'd rather handle these flags in two stages (pseudocode):

```cr
mode |= File::Mode::Write if mode.append?

flags = 0
if mode.read? && mode.write?
  flags |= RDWR
elsif mode.read?
  flags |= RDONLY
elsif mode.write?
  flags |= WRONLY
else
  raise "Must specify one of ..."
end

flags |= O_CREAT if mode.create?
...etc
```

and there's no need for the mode.each loop, in fact the code is shorter (and more easily optimizable) without.

Reply:
The loop is to prevent unhandled modes from slipping through by later additions.  Are you sure you want it removed?

Reply:
It's up to you, but I think the loop will be slower and specs will cover unhandled modes (who will add a mode without testing it and writing a spec??)

The handling for the read/write flags is much better though.